crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -102,12 +89,22 @@ impl<Wg: WireGuard, E: Backoff> StunEndpointProvider<Wg, E> {
                 ping_pong_tracker: ping_pong_handler,
                 stun_peer_publisher,
                 stun_state: StunState::WaitingForWg,
+                sockets: None,
             }),
         }
     }
 
-    pub async fn configure(&self, mut servers: Vec<Server>) {
+    pub async fn configure(&self, mut servers: Vec<Server>, sockets: Option<StunSockets>) {
         let _ = task_exec!(&self.task, async move |s| {
+            if let Some(stun_sockets) = sockets.as_ref() {
+                telio_log_debug!(
+                    "tun_socket: {:?}, ext_socket: {}",
+                    stun_sockets.tun_socket.as_native_socket(),
+                    stun_sockets.ext_socket.deref().as_native_socket()
+                );
+            }
+            s.sockets = sockets;
```

Comment:
The override of the sockets, means that any previous STUN session which we may have had are now invalid. Thus we should invalidate existing session and change the state of the STUN to the initial state. Just we must be careful not to do that on _every_ `configure()` call, but only on those, where we are actually transitioning from ipv6->ipv4 or vice-versa.

This applies to the sockets override as well. As new sockets will add new source port, meaning that new NAT.mapping will be created for that.